projects/github-actions/repo-gardening/src/tasks/gather-support-references/index.js

```diff
@@ -98,38 +100,81 @@ async function createOrUpdateComment( payload, octokit, issueReferences, issueCo
 	const { name: repo, owner } = repository;
 	const ownerLogin = owner.login;
 
-	const comment = `**Support References**
-	${ issueReferences
-		.map(
-			reference => `
-- [ ] ${ reference }`
-		)
-		.join( '' ) }
-
-	`;
-
-	const commentOpts = {
-		owner: ownerLogin,
-		repo,
-		body: comment,
-	};
-
 	const existingComment = await getListComment( issueComments );
 
 	// If there is a comment already, update it.
 	if ( existingComment !== 0 ) {
 		debug(
 			`gather-support-references: update comment ID ${ existingComment } with our new list of references.`
 		);
+
+		const {
+			data: { body: existingCommentBody },
+		} = await octokit.rest.issues.getComment( {
+			owner: ownerLogin,
+			repo,
+			comment_id: existingComment,
+		} );
+
+		// First, build a list of all references and whether they are checked or not.
+		const listWithStatusMatch = existingCommentBody.matchAll(
+			/(?:-\s\[(?<checked>\s|x)\]\s)(?<ticketId>[0-9]*-(?:hc|zen))/gm
+		);
+		const statusMatches = [ ...listWithStatusMatch ];
```

Comment:
No need for this, you can use `listWithStatusMatch` directly in the for-of loop below.